### PR TITLE
Added NATS Result Backend

### PR DESCRIPTION
**Pull request by chandr-andr**

### CODE REVIEW

**Comment by HemangLavana** [comment]
what is the status of this PR? I just came across taskiq-python module and wanted to use it with nats. However, was surprised to find that it required redis for results storage. I would prefer to use one system like nats (since it already supports key-value store) and not have to worry about maintaining two systems like nats and redis.

I hope this can PR can be merged soon and made part of official release. In the meantime, I will try it out locally to see if it works for my use case.

**Comment by s3rius** [comment]
@chandr-andr, can you please check this one?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -22,37 +22,19 @@ repos:
...
-        name: Check with Flake8
-        entry: poetry run flake8
+      - id: ruff
+        name: Run ruff lints
+        entry: poetry run ruff
         language: system
         pass_filenames: false
         types: [python]
-        args: [--count, taskiq_nats, tests]
+        args:
```
Comment (s3rius) [inline .pre-commit-config.yaml:31]: Add `chech` arg here.

**Comment by chandr-andr** [comment]
@s3rius It's waiting for you review)

File: taskiq_nats/result_backend.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            )
+        except ObjectNotFoundError as exc:
+            raise ResultIsMissingError from exc
+
+        if not self.keep_results:
+            await self.object_store.delete(
+                name=task_id,
+            )
+
+        taskiq_result: TaskiqResult[_ReturnType] = pickle.loads(  # noqa: S301
```
Comment (s3rius) [inline taskiq_nats/result_backend.py:115]: Can you replace pickle with TaskiqSerializers as [here](https://github.com/taskiq-python/taskiq-redis/blob/65a0b6c431e8cb256817f4ab6455d9ab043078f6/taskiq_redis/redis_backend.py#L120).

**Review by s3rius — APPROVED** [review]
Have two small comments. But everything else looks good.

**Pull request closed without merging**